4985_CommAudio/server.cpp

```diff
@@ -3,9 +3,159 @@
 #include <map>
 #include <WS2tcpip.h>
 
+QAudioFormat format;
+QAudioOutput *output;
+QBuffer *audioBuffer;
+QString audioFilePath;
```

Comment:
Couple things about audioFilePath: Why does it need to be global?  Why is this variable even necessary?  It looks like you set it to the filepath that is passed in `playAudio(QString &filepath)` and you don't use it anywhere else.  Can you not just use filepath?